pkg/cmd/pulumi/ai_web.go

```diff
@@ -0,0 +1,154 @@
+)
+
+func (l *PulumiAILanguage) String() string {
+	return string(*l)
+}
+
+func (l *PulumiAILanguage) Set(v string) error {
+	switch strings.ToLower(v) {
+	case "typescript", "ts":
+		*l = TypeScript
+	case "python", "py":
+		*l = Python
+	case "go", "golang":
+		*l = Go
+	case "c#", "csharp":
+		*l = CSharp
+	case "java":
+		*l = Java
+	case "yaml":
+		*l = YAML
+	default:
+		return fmt.Errorf("invalid language %q", v)
+	}
+	return nil
+}
+
+func (l *PulumiAILanguage) Type() string {
+	return "string"
+}
+
+type aiWebCmd struct {
+	appURL            string
+	disableAutoSubmit bool
+	language          PulumiAILanguage
+
+	Stdout io.Writer // defaults to os.Stdout
+
+	// currentBackend is a reference to the top-level currentBackend function.
+	// This is used to override the default implementation for testing purposes.
+	currentBackend func(context.Context, *workspace.Project, display.Options) (backend.Backend, error)
+}
+
+func (cmd *aiWebCmd) Run(ctx context.Context, args []string) error {
+	if cmd.Stdout == nil {
+		cmd.Stdout = os.Stdout
+	}
+
+	if cmd.currentBackend == nil {
+		cmd.currentBackend = currentBackend
+	}
+	requestURL, err := url.Parse(cmd.appURL)
+	if err != nil {
+		return err
+	}
+	query := requestURL.Query()
+	if len(args) > 0 {
+		query.Set("prompt", args[0])
+	}
+	if !cmd.disableAutoSubmit {
+		if len(args) == 0 {
+			return fmt.Errorf("prompt must be provided when auto-submit is enabled")
+		}
+		query.Set("autoSubmit", "true")
+	}
+	query.Set("language", cmd.language.String())
+
+	requestURL.RawQuery = query.Encode()
+	err = browser.OpenURL(requestURL.String())
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func newAIWebCommand() *cobra.Command {
+	var aiwebcmd aiWebCmd
+	aiwebcmd.appURL = env.AIServiceEndpoint.Value()
+	if aiwebcmd.appURL == "" {
+		aiwebcmd.appURL = "https://www.pulumi.com/ai"
+	}
+	cmd := &cobra.Command{
+		Use:   "web",
+		Short: "Opens Pulumi AI in your local browser",
+		Long: `Opens Pulumi AI in your local browser
+
+This command opens the Pulumi AI web app in your local default browser.
+It can be further initialized by providing a prompt to pre-fill in the app,
+with the default behavior then automatically submitting that prompt to Pulumi AI.
+
+If no prompt is provided, the app will be opened with no prompt pre-filled.
+
+If you do not want to submit the prompt to Pulumi AI, you can opt-out of this
+by passing the --no-auto-submit flag.
+`,
+		Args: cmdutil.MaximumNArgs(1),
+		Run: cmdutil.RunFunc(func(cmd *cobra.Command, args []string) error {
+			ctx := commandContext()
+			return aiwebcmd.Run(ctx, args)
+		},
+		),
+	}
+	cmd.PersistentFlags().BoolVar(
+		&aiwebcmd.disableAutoSubmit, "no-auto-submit", false,
+		"Opt-out of automatically submitting the prompt to Pulumi AI",
+	)
+	cmd.PersistentFlags().VarP(
+		&aiwebcmd.language, "language", "l",
+		"Language to use for the prompt - this defaults to TypeScript. [TypeScript, Python, Go, C#, Java, YAML]",
```

Comment:
Could we identify if the user is in a Pulumi project and if getting the project workspace succeeds, use that language by default?

Reply:
We certainly could - is that worth blocking the initial implementation for?

Reply:
No, I think not.